src/AppInstallerRepositoryCore/Microsoft/PinningIndex.cpp

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "PinningIndex.h"
+#include "SQLiteStorageBase.h"
+#include "Schema/Pinning_1_0/PinningIndexInterface.h"
+
+namespace AppInstaller::Repository::Microsoft
+{
+    PinningIndex PinningIndex::CreateNew(const std::string& filePath, Schema::Version version)
+    {
+        AICLI_LOG(Repo, Info, << "Creating new Pinning Index [" << version << "] at '" << filePath << "'");
+        PinningIndex result{ filePath, version };
+
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(result.m_dbconn, "pinningindex_createnew");
+
+        // Use calculated version, as incoming version could be 'latest'
+        result.m_version.SetSchemaVersion(result.m_dbconn);
+
+        result.m_interface->CreateTables(result.m_dbconn);
+
+        result.SetLastWriteTime();
+
+        savepoint.Commit();
+
+        return result;
+    }
+
+    std::shared_ptr<PinningIndex> PinningIndex::OpenOrCreateDefault(OpenDisposition openDisposition)
+    {
+        auto indexPath = Runtime::GetPathTo(Runtime::PathName::LocalState) / "pinning.db";
+        AICLI_LOG(CLI, Info, << "Opening pinning index");
+
+        try
+        {
+            if (std::filesystem::exists(indexPath))

```

Comment:
>if (std::filesystem::exists(indexPath)) [](http://example.com/codeflow?start=12&length=39)

I think it may be better if we do 
if (exists)
{
   if (isfile)
   {
       try open and return
   }
   
    delete and create new
}
else
{
     create new
}